web3_proxy/src/frontend/authorization.rs

```diff
@@ -1186,6 +1206,44 @@ impl Web3ProxyApp {
                                 None
                             };
 
+                        // Get the user_tier
+                        let user_model = user::Entity::find_by_id(rpc_key_model.user_id)
+                            .one(db_replica.conn())
+                            .await?
+                            .expect("related user");
+
+                        let balance = balance::Entity::find()
+                            .filter(balance::Column::UserId.eq(user_model.id))
+                            .one(db_replica.conn())
+                            .await?
+                            .expect("related balance")
+                            .available_balance;
+
+                        let mut user_tier_model =
+                            user_tier::Entity::find_by_id(user_model.user_tier_id)
+                                .one(db_replica.conn())
+                                .await?
+                                .expect("related user tier");
+
+                        // TODO: Do the logic here, as to how to treat the user, based on balance and initial check
+                        // Clear the cache (not the login!) in the stats if a tier-change happens (clear, but don't modify roles)
+                        if user_tier_model.title == UserTier::Premium.to_string()
+                            && balance < Decimal::from(10)
+                        {
+                            // Find the equivalent downgraded user tier, and modify limits from there
+                            if let Some(downgrade_user_tier) = user_tier_model.downgrade_tier_id {
+                                user_tier_model =
+                                    user_tier::Entity::find_by_id(downgrade_user_tier)
+                                        .one(db_replica.conn())
+                                        .await?
+                                        .expect("downgrade user tier for premium");
+                            } else {
+                                return Web3ProxyResult::Err(Web3ProxyError::InvalidUserTier);
+                            }
+                        }
+
+                        let user_tier = UserTier::try_from(user_tier_model.title.as_str())?;
```

Comment:
This won't work when we have a bunch of custom tiers. We don't need a custom model. Why not just use the actual user_tier database object?

Reply:
i removed this logic completely, see other comments for (possible) minor implications